kedro/config/omegaconf_config.py

```diff
@@ -445,6 +446,26 @@ def _resolve_environment_variables(config: dict[str, Any]) -> None:
         else:
             OmegaConf.resolve(config)
 
+    @staticmethod
+    def _destructive_merge(config, env_config, env_path):
+        # Destructively merge the two env dirs. The chosen env will override base.
+        common_keys = config.keys() & env_config.keys()
+        if common_keys:
+            sorted_keys = ", ".join(sorted(common_keys))
+            msg = (
+                "Config from path '%s' will override the following "
+                "existing top-level config keys: %s"
+            )
+            _config_logger.debug(msg, env_path, sorted_keys)
+
+        config.update(env_config)
+        return config
+
+    @staticmethod
+    def _soft_merge(config, env_config):
+        # Soft merge the two env dirs. The chosen env will override base if keys clash.
+        return OmegaConf.merge(config, env_config)
```

Comment:
I think this still needs to do 
```suggestion
        return OmegaConf.to_container(OmegaConf.merge(config, env_config))
```

Reply:
Ah ooops yes, I thought I had only removed `resolve=True`. Will update!